python/ray/util/tracing/tracing_helper.py

```diff
@@ -0,0 +1,443 @@
+                            attributes=_actor_hydrate_span_args(
+                                self.__class__.__name__, method),
+                        ):
+                    return method(self, *_args, **_kwargs)
+            else:
+                with tracer.start_as_current_span(
+                        _actor_span_consumer_name(self.__class__.__name__,
+                                                  method),
+                        kind=trace.SpanKind.CONSUMER,
+                        attributes=_actor_hydrate_span_args(
+                            self.__class__.__name__, method),
+                ):
+                    return method(self, *_args, **_kwargs)
+
+        return _resume_span
+
+    def async_span_wrapper(method: Callable[..., Any]) -> Any:
+        async def _resume_span(
+                self: Any,
+                *_args: Any,
+                _ray_trace_ctx: Optional[Dict[str, Any]] = None,
+                **_kwargs: Any,
+        ) -> Any:
+            """
+            Wrap the user's function with a function that
+            will extract the trace context
+            """
+            # If tracing feature flag is not on, perform a no-op
+            if not is_tracing_enabled():
+                return await method(self, *_args, **_kwargs)
+
+            tracer = trace.get_tracer(__name__)
+
+            # Retrieves the context from the _ray_trace_ctx dictionary we
+            # injected, or starts a new context
+            if _ray_trace_ctx:
+                with use_context(DictPropagator.extract(
+                        _ray_trace_ctx)), tracer.start_as_current_span(
+                            _actor_span_consumer_name(self.__class__.__name__,
+                                                      method.__name__),
+                            kind=trace.SpanKind.CONSUMER,
+                            attributes=_actor_hydrate_span_args(
+                                self.__class__.__name__, method.__name__),
+                        ):
+                    return await method(self, *_args, **_kwargs)
+            else:
+                with tracer.start_as_current_span(
+                        _actor_span_consumer_name(self._wrapped.__name__,
+                                                  method.__name__),
+                        kind=trace.SpanKind.CONSUMER,
+                        attributes=_actor_hydrate_span_args(
+                            self._wrapped.__name__, method.__name__),
+                ):
+                    return await method(self, *_args, **_kwargs)
+
+        return _resume_span
+
+    methods = inspect.getmembers(_cls, is_function_or_method)
+    for name, method in methods:
+        # Skip tracing for staticmethod or classmethod, because these method
+        # might not be called directly by remote calls. Additionally, they are
+        # tricky to get wrapped and unwrapped.
+        if is_static_method(_cls, name) or is_class_method(method):
```

Comment:
```suggestion
        if is_static_method(_cls, name) or is_class_method(method) or not is_tracing_enabled():
```

and remove it from `span_wrapper` and `async_span_wrapper`